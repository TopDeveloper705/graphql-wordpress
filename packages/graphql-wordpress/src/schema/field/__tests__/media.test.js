import {
  featuredMedia,
  description,
  caption,
  altText,
  mediaType,
  mimeType,
  sourceUrl,
} from 'field/media';
import getLoaders from 'data/loaders';

const args = {};
const context = {};
const info = {
  rootValue: {
    loaders: getLoaders(),
  },
};

describe('Test schema type field definition', () => {
  test('Test featuredMedia field', () => {
    const field = featuredMedia();
    expect(field.type.name).toMatchSnapshot();
  });

  test('Test featuredMedia field resolver', async () => {
    const field = featuredMedia();
    const media = await field.resolve({ featured_media: 69 }, args, context, info);
    expect(media).toMatchSnapshot();
  });

  test('Test featuredMedia field resolver null', () => {
    const field = featuredMedia();
    expect(field.resolve({ featured_media: 0 }, args, context, info)).toBeNull();
  });

  test('Test description field', () => {
    const { description: field } = description;
    expect(field.type).toMatchSnapshot();
  });

  test('Test caption field', () => {
    const { caption: field } = caption;
    expect(field.type.name).toMatchSnapshot();
  });

  test('Test altText field', () => {
    const { altText: field } = altText;
    expect(field.type).toMatchSnapshot();
  });

  test('Test mediaType field', () => {
    const { mediaType: field } = mediaType;
    expect(field.type.name).toMatchSnapshot();
  });

  test('Test mimeType field', () => {
    const { mimeType: field } = mimeType;
    expect(field.type).toMatchSnapshot();
  });

  test('Test sourceUrl field', () => {
    const { sourceUrl: field } = sourceUrl;
    expect(field.type).toMatchSnapshot();
  });
});
