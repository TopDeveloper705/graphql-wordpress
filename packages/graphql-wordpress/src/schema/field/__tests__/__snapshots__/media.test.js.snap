// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test schema type field definition Test altText field 1`] = `"String"`;

exports[`Test schema type field definition Test caption field 1`] = `"Caption"`;

exports[`Test schema type field definition Test description field 1`] = `"Description"`;

exports[`Test schema type field definition Test featuredMedia field 1`] = `"Media"`;

exports[`Test schema type field definition Test featuredMedia field resolver 1`] = `
Media {
  "id": 69,
}
`;

exports[`Test schema type field definition Test mediaType field 1`] = `"MEDIA_TYPE"`;

exports[`Test schema type field definition Test mimeType field 1`] = `"String"`;

exports[`Test schema type field definition Test sourceUrl field 1`] = `"String"`;
